suite-native/discovery/src/discoveryThunks.ts

```diff
@@ -1,24 +1,247 @@
+
+        if (isIncompatibleCardanoType || wasLimitReached) {
+            dispatch(finishNetworkTypeDiscoveryThunk());
+            return;
+        }
+
+        const chunkBundle: Array<DiscoveryItem & { showOnTrezor: boolean }> = [];
+
+        Array.from(Array(DISCOVERY_ROUND_SIZE).keys()).forEach(index => {
+            const accountPath = network.bip43Path.replace(
+                'i',
+                (lastDiscoveredAccountIndex + index).toString(),
+            );
+
+            // Check if account does not already exists.
+            const account = selectAccountByNetworkAndPath(getState(), network.symbol, accountPath);
+
+            if (!account) {
+                chunkBundle.push({
+                    path: accountPath,
+                    coin: network.symbol,
+                    index,
+                    accountType,
+                    networkType: network.networkType,
+                    derivationType: getDerivationType(accountType),
+                    suppressBackupWarning: true,
+                    showOnTrezor: false,
+                });
+            }
+        });
+
+        // All accounts of the batch were already discovered, skip it.
+        if (!chunkBundle) {
+            dispatch(
+                discoverNetworkBatchThunk({
+                    deviceState,
+                    device,
+                    network,
+                    mutex,
+                    round: round + 1,
+                }),
+            );
+        }
+
+        // Take exclusive access to the device and hold it until is the fetching of the descriptors done.
+        await mutex.lock();
+        const descriptorsBundle = await getBundleDescriptors(chunkBundle, network.networkType);
+        mutex.unlock();
+
+        const isFinished = await dispatch(
+            discoverAccountsByDescriptorThunk({
+                descriptorsBundle,
+                deviceState,
+            }),
+        ).unwrap();
+
+        if (!isFinished) {
+            dispatch(
+                discoverNetworkBatchThunk({
+                    deviceState,
+                    device,
+                    network,
+                    mutex,
+                    round: round + 1,
+                }),
+            );
+        } else {
+            dispatch(finishNetworkTypeDiscoveryThunk());
+        }
+    },
+);
+
+export const startDescriptorPreloadedDiscovery = createThunk(
+    `${DISCOVERY_MODULE_PREFIX}/startXpubPreloadedDiscovery`,
+    async (
+        { deviceState, device }: { deviceState: string; device: TrezorDevice },
+        { dispatch, getState },
+    ) => {
+        const mutex = new DeviceAccessMutex();
         await dispatch(
             createDiscoveryThunk({
                 deviceState,
                 device,
             }),
         );
+        const discovery = selectDiscoveryForDevice(getState());
+
+        if (!discovery) {
+            return;
+        }
+
+        dispatch({
```

Comment:
Please use the action creator that's available. This way, you have 0 type safety.

Reply:
fixed in 33d0b925d3